CMakeLists.txt

```diff
@@ -66,15 +66,32 @@ set(MINGW_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX}/${TARGET_ARCH} CACHE STRING "Ta
 set(SINGLE_SOURCE_LOCATION "" CACHE STRING "Repository or tarball stored path")
 set(RUSTUP_LOCATION "${CMAKE_CURRENT_BINARY_DIR}/install_rustup" CACHE STRING "Rust toolchain path")
 set(LLVM_ENABLE_LTO "OFF" CACHE STRING "OFF, ON, Thin and Full")
+set(LLVM_ENABLE_PGO "OFF" CACHE STRING "OFF, GEN, CSGEN, USE")
+if(LLVM_ENABLE_PGO STREQUAL "GEN" OR LLVM_ENABLE_PGO STREQUAL "CSGEN")
+    set(LLVM_PROFILE_DATA_DIR "${CMAKE_INSTALL_PREFIX}/profiles/" CACHE PATH "Default profile generation directory")
+endif()
+if(LLVM_ENABLE_PGO STREQUAL "USE" OR LLVM_ENABLE_PGO STREQUAL "CSGEN")
+    set(LLVM_PROFDATA_FILE "" CACHE FILEPATH "Profdata file path")
+    if(NOT (EXISTS ${LLVM_PROFDATA_FILE}))
+        set(LLVM_ENABLE_PGO OFF)
+    endif()
+endif()
 option(CLANG_PACKAGES_LTO "Build each package with ThinLTO." OFF)
 option(ENABLE_CCACHE "Enable ccache" OFF)
 if(ENABLE_CCACHE)
+    set(CCACHE_MAXSIZE "500M" CACHE STRING "Size of ccache")
+    set(CCACHE_DIR "${MINGW_INSTALL_PREFIX}/ccache" CACHE STRING "Directory to keep ccached data")
```

Comment:
I dont think `CCACHE_DIR ` option will meaningful as it cannot be shared across multiple target